concerned-coyotes/earlyinternet/news/models.py

```diff
@@ -0,0 +1,60 @@
+from typing import Dict, Union
+
+from django.db import models
+from django.utils.dateparse import parse_datetime
+
+
+class ArticleManager(models.Manager):
+
+    def create_article(self, raw_article: Dict[str, Union[str, Dict[str, str]]]):
+        """
+        Create a new article in the database from the json data given
+        by NewsApiClient.get_top_headlines['articles']
+
+        >>> Article.objects.create_article({...})
+
+        :param raw_article: Dictionary containing information about the article
+        :return: New instance of Article
+        """
+
+        # convert timezone aware iso date string to datetime.datetime
+        published_at = parse_datetime(raw_article['publishedAt'])
+
+        article = self.create(
+            source=raw_article['source']['name'],
+            author=raw_article['author'],
+            title=raw_article['title'],
+            description=raw_article['description'],
+            content=raw_article['content'],
+            url=raw_article['url'],
+            published_at=published_at
+        )
+        return article
+
+    def get_latest(self, n_items):
+        """
+        Return the latest n articles
+
+        :param n_items: How many articles to return
+        :return: List of n newest articles
+        """
+        return super().get_queryset().order_by("-published_at")[:n_items]
+
+
+class Article(models.Model):
```

Comment:
A lot of the models only seem to be queried in a particular order, like `"-published_at"` for `Article`. In this case. both the dashboard view as well as the `get_latest` method of the `ArticleManager` both specify that ordering.

Instead of having to specify that ordering each time, you could introduce a default ordering for models by adding a `Meta` class to your model.

Example:

```py
class MyModel(models.Model):
    # interesting fields here...
    published_at = models.DateTimeField()

    class Meta:
        ordering = ["-published_at"]
```